device/inventec/x86_64-inventec_d7054q28b-r0/plugins/sfputil.py

```diff
@@ -0,0 +1,210 @@
+        except IOError as e:
+            print "Error: unable to open file: %s" % str(e)
+            return False
+
+        reg_value = int(reg_file.readline().rstrip())
+
+        if reg_value == 0:
+            return True     
+
+        return False
+
+    def get_low_power_mode(self, port_num):
+        # Check for invalid port_num
+        if port_num < self.port_start or port_num > self.port_end:
+            return False
+        if port_num < self.qsfp_port_start or port_num > self.qsfp_port_end:
+            return False
+
+        try:
+            reg_file = open("/sys/class/swps/port"+str(port_num)+"/lpmod")
+        except IOError as e:
+            print "Error: unable to open file: %s" % str(e)
+
+        reg_value = int(reg_file.readline().rstrip())
+
+        if reg_value == 0:
+            return False       
+
+        return True
+
+    def set_low_power_mode(self, port_num, lpmode):
+        # Check for invalid port_num
+        if port_num < self.port_start or port_num > self.port_end:
+            return False
+        if port_num < self.qsfp_port_start or port_num > self.qsfp_port_end:
+            print "\nError:SFP's don't support this property"
+            return False
+
+        try:
+            reg_file = open("/sys/class/swps/port"+str(port_num)+"/lpmod", "r+")
+        except IOError as e:
+            print "Error: unable to open file: %s" % str(e)
+            return False
+
+        reg_value = int(reg_file.readline().rstrip())
+               
+        # LPMode is active high; set or clear the bit accordingly
+        if lpmode is True:
+            reg_value = 1
+        else:
+            reg_value = 0   
+        
+        reg_file.write(hex(reg_value))
+        reg_file.close()
+
+        return True
+
+    def reset(self, port_num):
+        QSFP_RESET_REGISTER_DEVICE_FILE = "/sys/class/swps/port"+str(port_num)+"/reset"
+        # Check for invalid port_num
+        if port_num < self.port_start or port_num > self.port_end:
+            return False
+        if port_num < self.qsfp_port_start or port_num > self.qsfp_port_end:
+            print "\nError:SFP's don't support this property"
+            return False
+
+        try:
+            reg_file = open(QSFP_RESET_REGISTER_DEVICE_FILE, "r+")
+        except IOError as e:
+            print "Error: unable to open file: %s" % str(e)
+            return False
+
+        reg_value = 0
+        reg_file.write(hex(reg_value))
+        reg_file.close()
+
+        # Sleep 2 second to allow it to settle
+        time.sleep(2)
+
+        # Flip the value back write back to the register to take port out of reset
+        try:
+            reg_file = open(QSFP_RESET_REGISTER_DEVICE_FILE, "r+")
+        except IOError as e:
+            print "Error: unable to open file: %s" % str(e)
+            return False
+
+        reg_value = 1
+        reg_file.write(hex(reg_value))
+        reg_file.close()
+
+        return True
+
```

Comment:
Remove extra trailing newline at EOF (file should end in one '\n')

Reply:
Hello Joe,

Please let me know the updates are fine.

Thanks and regards,
Bala

Reply:
Hello Joe,

Sorry I misunderstood.
Please confirm the changes.

Thanks and regards,
Bala